app/src/main/java/com/duckduckgo/app/browser/omnibar/experiments/FadeOmnibarLayout.kt

```diff
@@ -152,27 +151,35 @@ class FadeOmnibarLayout @JvmOverloads constructor(
     }
 
     override fun render(viewState: ViewState) {
-        if (viewState.viewMode is ViewMode.CustomTab) {
+        super.render(viewState)
+        outlineProvider = if (viewState.viewMode is ViewMode.CustomTab) {
             // adds a drop shadow for the AppBarLayout, in case it was removed at any point
-            outlineProvider = ViewOutlineProvider.BACKGROUND
-            super.render(viewState)
+            ViewOutlineProvider.BACKGROUND
         } else {
             // removes the drop shadow from the AppBarLayout to make it appear flat in the view hierarchy
-            outlineProvider = null
+            null
+        }
 
-            // removes the duplicate buttons that are also present in the navigation bar
-            val experimentalViewState = viewState.copy(
-                showBrowserMenu = false,
-                showFireIcon = false,
-                showTabsMenu = false,
-            )
-            super.render(experimentalViewState)
+        if (viewState.hasFocus || isFindInPageVisible) {
+            animateOmnibarFocusedState(focused = true)
+        } else {
+            animateOmnibarFocusedState(focused = false)
         }
+    }
 
-        aiChat.isVisible = viewState.showChat
-        aiChatDivider.isVisible = (viewState.showVoiceSearch || viewState.showClearButton) && viewState.showChat
+    override fun renderButtons(viewState: ViewState) {
+        tabsMenu.isVisible = false
+        fireIconMenu.isVisible = false
+        browserMenu.isVisible = viewState.viewMode is ViewMode.CustomTab
+        browserMenuHighlight.isVisible = false
+        clearTextButton.isVisible = viewState.showClearButton
+        voiceSearchButton.isVisible = viewState.showVoiceSearch
         spacer.isVisible = false
 
+        val showAiChat = shouldShowExperimentalAIChatButton(viewState)
+        aiChatMenu?.isVisible = showAiChat
+        aiChatDivider.isVisible = (viewState.showVoiceSearch || viewState.showClearButton) && showAiChat
```

Comment:
Can `aiChatMenu` actually ever be `null`? If yes, we should also make that a condition to show the divider. If no, let's remove the nullability from the type.

Reply:
it can, it has to do with the animations that @joshliebe created. We can leave it as is for now.